CMakeModules/FindMKL.cmake

```
@@ -62,7 +62,12 @@ include(CheckTypeSize)
 check_type_size("int" INT_SIZE
   BUILTIN_TYPES_ONLY LANGUAGE C)
 
-set(MKL_THREAD_LAYER "Sequential" CACHE STRING "The thread layer to choose for MKL")
+
+if(NOT MKL_THREAD_LAYER)
+    set(MKL_THREAD_LAYER "TBB")
+endif()
+
+set(MKL_THREAD_LAYER ${MKL_THREAD_LAYER} CACHE STRING "The thread layer to choose for MKL")
```

```suggestion
set(MKL_THREAD_LAYER "TBB" CACHE STRING "The thread layer to choose for MKL")
```
User's choice will always overwrite.

---

Why did it not work before then?

---

That line was exactly my problem.

---

Before the default was `Sequential`, right?

---

Yes but in tools CMakeList the parameter was explicitly set

---

But FindMKL did not use it

---

I see, `set(MKL_THREAD_LAYER "Intel OpenMP")` isn't enough, you need to do something like `set(MKL_THREAD_LAYER "Intel OpenMP" CACHE STRING ...)` there.